[BUG][okhttp-gson] can not generate javadoc

##### Description

Introduced with issue #2994 / PR #2995 the generated javadoc now contains `@http.response.details`. Example from our the samples:

https://github.com/OpenAPITools/openapi-generator/blob/51e7005373fdb55daa1c253f36fb6267a64a7bf7/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/api/StoreApi.java#L63-L68

When trying to generate the javadoc artifact for the project, you get tons of errors:

```
cd samples/client/petstore/java/okhttp-gson
mvn javadoc:javadoc
```

```
INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.419 s
[INFO] Finished at: 2019-07-08T10:44:48+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:javadoc (default-cli) on project petstore-okhttp-gson: An error has occurred in JavaDocs report generation: 
[ERROR] Exit code: 1 - /____/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/api/AnotherFakeApi.java:63: error: unknown tag: http.response.details
[ERROR]      * @http.response.details
[ERROR]        ^
[ERROR] /____/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/api/AnotherFakeApi.java:117: error: unknown tag: http.response.details
[ERROR]      * @http.response.details
[ERROR]        ^
[ERROR] /____/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/api/AnotherFakeApi.java:134: error: unknown tag: http.response.details
[ERROR]      * @http.response.details
[ERROR]        ^
```

##### openapi-generator version

4.0.2

##### Suggest a fix

We can keep the table, but remove the `@http.response.details` and just put "Response details:"

@saigiridhar21, @wing328 what do you think?

--- a/CI/circle_parallel.sh
+++ b/CI/circle_parallel.sh
@@ -12,6 +12,7 @@ if [ "$NODE_INDEX" = "1" ]; then
   #cp CI/pom.xml.circleci pom.xml
   java -version
   mvn --quiet verify -Psamples.circleci
+  mvn --quiet javadoc:javadoc -Psamples.circleci
 
   # generate all petstore samples (client, servers, doc)
   ./bin/run-all-petstore

--- a/modules/openapi-generator/src/main/resources/Java/libraries/jersey2/build.gradle.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/jersey2/build.gradle.mustache
@@ -161,3 +161,7 @@ dependencies {
     {{/java8}}
     testCompile "junit:junit:$junit_version"
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/modules/openapi-generator/src/main/resources/Java/libraries/jersey2/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/jersey2/pom.mustache
@@ -170,17 +170,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/build.gradle.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/build.gradle.mustache
@@ -143,3 +143,7 @@ dependencies {
     {{/threetenbp}}
     testCompile 'junit:junit:4.12'
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pom.mustache
@@ -149,17 +149,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/modules/openapi-generator/src/main/resources/java-pkmst/config/appconfig.mustache
+++ b/modules/openapi-generator/src/main/resources/java-pkmst/config/appconfig.mustache
@@ -87,7 +87,7 @@ public class AppConfig extends WebMvcConfigurerAdapter {
 	 * .exposedHeaders("header1", "header2")
 	 * .allowCredentials(false).maxAge(3600);
 	 * 
-	 * @return
+	 * @return a new WebMvcConfigurer instance
 	 */
 	@Bean
 	public WebMvcConfigurer corsConfigurer() {

--- a/modules/openapi-generator/src/main/resources/java-pkmst/security/oAuth2SecurityConfiguration.mustache
+++ b/modules/openapi-generator/src/main/resources/java-pkmst/security/oAuth2SecurityConfiguration.mustache
@@ -18,7 +18,7 @@ import org.springframework.security.oauth2.provider.request.DefaultOAuth2Request
 import org.springframework.security.oauth2.provider.token.TokenStore;
 import org.springframework.security.oauth2.provider.token.store.InMemoryTokenStore;
 /**
- * Provides a convenient base class for creating a {@link WebSecurityConfigurer}
+ * Provides a convenient base class for creating a WebSecurityConfigurer
  * instance. The implementation allows customization by overriding methods.
  *
  * @see EnableWebSecurity

--- a/modules/openapi-generator/src/main/resources/java-pkmst/security/resourceServerConfiguration.mustache
+++ b/modules/openapi-generator/src/main/resources/java-pkmst/security/resourceServerConfiguration.mustache
@@ -10,7 +10,7 @@ import org.springframework.http.HttpMethod;
 /**
  * Configurer class for <code>@EnableResourceServer</code> classes. This class adjust the access
  * rules and paths that are protected by OAuth2 security. If more than one configures the same property, then the last
- * one wins. The configurers are sorted by {@link Order} before being applied.
+ * one wins. The configurers are sorted by Order before being applied.
  * 
  * @author pkmst
  * 

--- a/samples/client/petstore/java/jersey2-java6/pom.xml
+++ b/samples/client/petstore/java/jersey2-java6/pom.xml
@@ -151,17 +151,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/samples/client/petstore/java/jersey2-java8/build.gradle
+++ b/samples/client/petstore/java/jersey2-java8/build.gradle
@@ -112,3 +112,7 @@ dependencies {
     compile "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jackson_version"
     testCompile "junit:junit:$junit_version"
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/samples/client/petstore/java/jersey2-java8/pom.xml
+++ b/samples/client/petstore/java/jersey2-java8/pom.xml
@@ -151,17 +151,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/samples/client/petstore/java/jersey2/build.gradle
+++ b/samples/client/petstore/java/jersey2/build.gradle
@@ -114,3 +114,7 @@ dependencies {
     compile "com.brsanthu:migbase64:2.2"
     testCompile "junit:junit:$junit_version"
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/samples/client/petstore/java/jersey2/pom.xml
+++ b/samples/client/petstore/java/jersey2/pom.xml
@@ -151,17 +151,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/samples/client/petstore/java/okhttp-gson-parcelableModel/build.gradle
+++ b/samples/client/petstore/java/okhttp-gson-parcelableModel/build.gradle
@@ -108,3 +108,7 @@ dependencies {
     compile 'org.threeten:threetenbp:1.3.5'
     testCompile 'junit:junit:4.12'
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/samples/client/petstore/java/okhttp-gson-parcelableModel/pom.xml
+++ b/samples/client/petstore/java/okhttp-gson-parcelableModel/pom.xml
@@ -142,17 +142,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/samples/client/petstore/java/okhttp-gson/build.gradle
+++ b/samples/client/petstore/java/okhttp-gson/build.gradle
@@ -108,3 +108,7 @@ dependencies {
     compile 'org.threeten:threetenbp:1.3.5'
     testCompile 'junit:junit:4.12'
 }
+
+javadoc {
+    options.tags = [ "http.response.details:a:Http Response Details" ]
+}

--- a/samples/client/petstore/java/okhttp-gson/pom.xml
+++ b/samples/client/petstore/java/okhttp-gson/pom.xml
@@ -142,17 +142,17 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
-                        <configuration>
-                            <tags>
-                                <tag>
-                                    <name>http.response.details</name>
-                                    <placement>a</placement>
-                                    <head>Http Response Details:</head>
-                                </tag>
-                 	        </tags>
-                        </configuration>
                     </execution>
                 </executions>
+                <configuration>
+                    <tags>
+                        <tag>
+                            <name>http.response.details</name>
+                            <placement>a</placement>
+                            <head>Http Response Details:</head>
+                        </tag>
+                    </tags>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/config/AppConfig.java
+++ b/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/config/AppConfig.java
@@ -20,7 +20,7 @@ public class AppConfig extends WebMvcConfigurerAdapter {
 	 * .exposedHeaders("header1", "header2")
 	 * .allowCredentials(false).maxAge(3600);
 	 * 
-	 * @return
+	 * @return a new WebMvcConfigurer instance
 	 */
 	@Bean
 	public WebMvcConfigurer corsConfigurer() {

--- a/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/security/OAuth2SecurityConfiguration.java
+++ b/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/security/OAuth2SecurityConfiguration.java
@@ -18,7 +18,7 @@ import org.springframework.security.oauth2.provider.request.DefaultOAuth2Request
 import org.springframework.security.oauth2.provider.token.TokenStore;
 import org.springframework.security.oauth2.provider.token.store.InMemoryTokenStore;
 /**
- * Provides a convenient base class for creating a {@link WebSecurityConfigurer}
+ * Provides a convenient base class for creating a WebSecurityConfigurer
  * instance. The implementation allows customization by overriding methods.
  *
  * @see EnableWebSecurity

--- a/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/security/ResourceServerConfiguration.java
+++ b/samples/server/petstore/java-pkmst/src/main/java/com/prokarma/pkmst/security/ResourceServerConfiguration.java
@@ -10,7 +10,7 @@ import org.springframework.http.HttpMethod;
 /**
  * Configurer class for <code>@EnableResourceServer</code> classes. This class adjust the access
  * rules and paths that are protected by OAuth2 security. If more than one configures the same property, then the last
- * one wins. The configurers are sorted by {@link Order} before being applied.
+ * one wins. The configurers are sorted by Order before being applied.
  * 
  * @author pkmst
  * 

